packages/plugin-infra/src/types.ts

```diff
@@ -69,7 +76,11 @@ export namespace Plugin.Shared {
         personaSign(payload: PersonaSignRequest): Promise<PersonaSignResult>
         /** Sign a message with wallet */
         walletSign(message: string, address: string): Promise<string>
+        /** Sign a message silently */
+        silentSign(signer: ECKeyIdentifier, message: string): Promise<PersonaSignResult>
```

Comment:
This looks dangerous. I don't want to expose this ability to all plugins.

Reply:
But I need this ability in web3 profile plugin

Reply:
Rename it to `priviliged_silentSign` and only allow `plugin ID = web3 profile` plugin to use the implementation provided.

```js
priviliged_silentSign: async () => {
    if (pluginID !== PluginID.Web3Profile) throw new TypeError(...)
}
```

Reply:
can I make it like this:
```Javascript
priviliged_silentSign: pluginID === PluginId.Web3Profile ? Services.Identity.generateSignResult
 : null,
```